### ISSUE TITLE
Checkpoint conversion

### ISSUE THREAD

**Issue by MaxiBoether** [pytorch/torchtitan#758]
Hey,

I am trying to evaluate a model trained with torchtitan using the lm eval harness. I am using the VLLM backend. Is there any straightforward way to convert a torchtitan model in the pytorch .pt format to, e.g., a huggingface model to be used in VLLM/lm eval harness? Within the torchtune repo, I was able to find [some code for VLMs](https://github.com/pytorch/torchtune/blob/main/recipes/eleuther_eval.py), but (a) that seems to be hardcoded for LLMs, (b) uses a new inference backend instead of e.g. relying on VLLM, and (c) I feel like there might be an easy way to convert torchtitan checkpoints rather than coming up with such an involved solution.

How did you evaluate downstream task accuracy with torchtitan models?

Thank you very much for your help.

**Comment by fegin** [comment]
When you save PyTorch format, is that full tensor (non-DTensor) and is saved with torch.save()?

**Comment by MaxiBoether** [comment]
I mean what the [checkpoint.md](https://github.com/pytorch/torchtitan/blob/main/docs/checkpoint.md) docs of torchtitan call `dcp_to_torch`. So the .pile generated from that, which is not - at least natively - supported by hf.

**Comment by jaysonfrancis** [comment]
You can take take a look at [convert_llama_weights_to_hf.py](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/convert_llama_weights_to_hf.py#L432)
